main.tf

```diff
@@ -18,14 +52,14 @@ resource "azurerm_management_lock" "this" {
 
   lock_level = var.lock.kind
   name       = coalesce(var.lock.name, "lock-${var.name}")
-  scope      = azurerm_TODO_resource.this.id
+  scope      = azurerm_kubernetes_cluster.this.id
 }
 
 resource "azurerm_role_assignment" "this" {
   for_each = var.role_assignments
```

Comment:
You are not testing for real this resource because `var.role_assignments` is an empty map.

Look at the necessary role assignments in https://github.com/Azure/terraform-azurerm-aks

Reply:
Here: https://github.com/Azure/terraform-azurerm-aks/blob/main/role_assignments.tf

Those have to be ported into the `var.role_assignments` map I believe.

If in doubt double check with @lonegunmanb

Reply:
for this PR i left out this implementation, but i can add this back